### PR TITLE
Version One Space Travelers

### PR DESCRIPTION
**Pull request by petudeveloper**
For this Pull request we add the following features:

- Project's setup
- Header
- Project's structure - Rockets
- Project's structure - Missions
- Rockets fetch
- Missions fetch
- Display rockets
- Rockets booking
- Cancel rocket
- Display missions
- Join/leave missions
- Switch badge - Rockets
- Switch badge - Missions
- Profile list user's Missions
- Profile list user's Rockets
- Unit Tests

### CODE REVIEW

File: src/components/missions/joinButton.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { joinMission, leaveMission } from '../../redux/missions/missions';
+import style from './missions.css';
+
+const JoinButton = ({ status, id }) => {
+  const dispatch = useDispatch();
+  const fields = {
+    'Not a member': {
+      variant: 'outline-secondary',
+      message: 'Join Mission',
+      click: () => { dispatch(joinMission(id)); },
```
Comment (e71az) [inline src/components/missions/joinButton.js:13]: - [ ] You should handle this as you did with the rockets, a reserved key should appear with its value set to true or false accordingly to comply with the project requirements, please do so.

The selected mission will have an extra key reserved with its value set to true (As you did for the rockets), example:
``` javascript
  const newState = state.map(mission => {
    if(mission.id !== id) 
        return mission;
    return { ...mission, reserved: true };
});
```

File: src/components/missions/joinButton.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    },
+    'Active member': {
+      variant: 'outline-danger',
+      message: 'Leave Mission',
+      click: () => { dispatch(leaveMission(id)); },
+    },
+  };
```
Comment (e71az) [inline src/components/missions/joinButton.js:20]: - [ ] Rockets/Dragons and Missions should use the React conditional rendering syntax, please do so to comply with the project requirements:

```javascript
{mission.reserved && ( 
    // render Leave Mission button
)}
```

File: src/pages/Profile.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  const missions = useSelector((state) => state.missions, shallowEqual).filter(({ status }) => status === 'Active member');
+
+  return (
+    <div className={style.profilecontainer}>
+      <section className={style.profilesection}>
+        <h3>My Rockets</h3>
+        {rockets && (
+        <ul className={style.profilelist}>
+          {
+                      rockets.map(({ name, id }) => (
```
Comment (e71az) [inline src/pages/Profile.js:17]: - [ ] Render a list of all joined missions and rockets.

* This should be done using `filter();` (Currently, map is being used for the rendering, please correct this to comply with the project requirements)

File: src/pages/Profile.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        {rockets.length === 0
+          && (
+          <Alert variant="secondary">
+            <Alert.Heading>No Rockets Reserved</Alert.Heading>
+            <p>
+              Go to the Rockets page and reserve one.
+              {' '}
+              <NavLink exact to="/">Click Here </NavLink>
+              {' '}
+              if you want to go the Rockets page.
```
Comment (e71az) [inline src/pages/Profile.js:34]: - This feature was optional but you added it, great job!

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @petudeveloper @leonardoeng13, I am Elias, your reviewer for this PR

Well done so far with your project!

Your project looks and works amazing, I really liked the animation that you placed to the status tag in the missions page! There are some really small details to be fixed in order to continue, keep it up!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [x] Please fix this output when your tests are run, I get a really big list of errors but the problem seems to be the same one, so fixing it in all your test scripts should be enough:

image

- [x] Please check the comments below in this PR for more requirements.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

1. The `My Profile` section can be enhanced by adding the "Cancel reservation" and "Leave Mission" buttons to the lists here. Clicking them should dispatch the actions you have already used in the main Rockets and Missions sections.
2. The `My Profile` section can be enhanced by adding the "Read more" button for each mission and rocket. Upon click, it should open a corresponding Wikipedia page in a new tab. NOTE - you need to get that extra Wikipedia URL from the API's payload.

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/pages/Profile.js (hunk 3 above)
Comment (petudeveloper) [reply src/pages/Profile.js:17]: @e71az The `filter();` is used on line 7:
 `const rockets = useSelector((state) => state.rockets).filter(({ reserved }) => reserved);`

File: src/components/missions/joinButton.js (hunk 1 above)
Comment (petudeveloper) [reply src/components/missions/joinButton.js]: Thanks for the feedback, functionality is updated.

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## Thank you for implementing all the project requirements, you've done a great job :trophy:  . There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**